### ISSUE TITLE
Geth should handle genesis block with default syncmode

### ISSUE THREAD

**Issue by quickchase** [ethereum/go-ethereum#28222]
#### System information

Geth version: [v1.13.2](https://github.com/ethereum/go-ethereum/tree/v1.13.2)

#### Expected behaviour
When running without specifying `--syncmode=full` (i.e. the default i.e. snap sync) geth should transition to full sync automatically for pre-genesis / genesis blocks.

#### Actual behaviour
Geth doesn't import the 1st block:

> Ignoring payload while snap syncing number=1

#### Steps to reproduce the behaviour

Run geth without `--syncmode=full` on a new network

**Comment by MariusVanDerWijden** [comment]
I don't understand this issue. You are running with the default sync mode which is snap, and then you want to switch to a full sync? What are pre-genesis blocks?

**Comment by quickchase** [comment]
I started geth on Holesky before it launched, before the genesis block...

Geth failed to start syncing the chain at genesis until I manually changed the sync mode to full.

https://twitter.com/parithosh_j/status/1707387712502689958?t=aIsu1zYVUdVa9gdGrE52OQ&s=19

**Comment by MariusVanDerWijden** [comment]
Ahh I understand, so geth kinda errors on snap sync if the genesis time has not been reached yet

**Comment by quickchase** [comment]
Yes, correct, and if genesis time is reached while already running it will attempt to snap sync but be unable to do so.

Geth patiently waits for genesis, but doesn't go anywhere when genesis finally happenes.

**Comment by Tanishka-47** [comment]
Hi team! 👋

I would like to work on this issue. I'm currently setting up my local environment and diving into the codebase to reproduce the issue and work on a fix.

Please assign this to me if it's still available. Thank you!
